zebra-state/src/service/tests.rs

```diff
@@ -183,3 +187,172 @@ fn state_behaves_when_blocks_are_committed_out_of_order() -> Result<()> {
 
     Ok(())
 }
+
+/*
+#[test]
```

Comment:
The ticket was missing a test for this rule:
> for every block, if the block height is less than the NU5 activation height, return Ok(())

Here's the test:
- proptest a block vector containing at least 100 blocks, starting less than 100 blocks past the NU5 activation height
  - some blocks must be lower than the NU5 activation height
  - the transactions need to all be V4, use the transaction version override from #2347

I've also added the missing test to the ticket, and tried to clarify some of the other tests.